### ISSUE TITLE
Investigate why we can't use both default and fill_existing_with

### ISSUE THREAD

**Issue by jwoertink** [luckyframework/avram#598]
https://github.com/luckyframework/avram/blob/b44793857899a06b4c3b9d45cdd60f0164ec9fd9/src/avram/migrator/alter_table_statement.cr#L143

I'm sure there was some technical reason for this. I didn't look too deep in to the code, but I would assume this should work. Imagine you're adding a column to your DB that's not nilable, you obviously want to fill the existing rows with some data since it can't be nil. But also imagine that this new column has some default value set to it. When you add a new row, you want that default value to be set automatically. 

This is the difference between these:

```crystal
column new_row : String
# vs
column new_row : String = "blamo"
```

**Comment by matthewmcgarvey** [comment]
Yea, I have no idea why you can't do both. And it looks like once you have a non-nilable column with a database-provided value you have to add `autogenerated: true` https://github.com/luckyframework/avram/blob/b44793857899a06b4c3b9d45cdd60f0164ec9fd9/src/avram/model.cr#L223. Hopefully that's correct!

**Comment by jwoertink** [comment]
I figured out the reason. It's been updated on the website https://github.com/luckyframework/website/issues/718

TL;DR - Postgres handles the backfill for us automagically. `default` will do both `fill_existing_with` AND set the default for new records.
